primitives/src/analytics.rs

```diff
@@ -0,0 +1,78 @@
+use crate::DomainError;
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct AnalyticsResponse {
+    time: u32,
+    value: String,
+}
+
+#[cfg(feature = "postgres")]
+pub mod postgres {
+    use super::AnalyticsResponse;
+    use tokio_postgres::Row;
+
+    impl From<&Row> for AnalyticsResponse {
+        fn from(row: &Row) -> Self {
+            Self {
+                time: row.get("time"),
+                value: row.get("value"),
+            }
+        }
+    }
+}
+
+#[derive(Debug, Deserialize)]
+pub struct AnalyticsQuery {
+    #[serde(default = "default_limit")]
+    pub limit: u32,
+    #[serde(default = "default_event_type")]
+    pub event_type: String,
+    #[serde(default = "default_metric")]
+    pub metric: String,
+    #[serde(default = "default_timeframe")]
+    pub timeframe: String,
+}
+
+impl AnalyticsQuery {
+    pub fn is_valid(&self) -> Result<(), DomainError> {
```

Comment:
Instead of using strings, it will be better to use Enums and leave this kind of validation to Serde.
Probably also we don't want a big limit as it might cause the DB to hang, we should think of a good max limit to put here.

Reply:
I don't think using enum offer any benefits at all. 
We would have to derive display so to call `to_string` on the enum, then the errors `serde` would return wouldn't make a lot of sense so we have to implement custom errors. Seem like an unneeded implementation detail

Reply:
Also, 100-200 (max) is a good limit. Postgres shouldn't hang on that if we have good indexes

Reply:
yes, but since this comes from deserialization of the query passed from the URL, you could pass 1 million and then it will hang, i.e. we need validation of the limit.

Reply:
1 million wouldn't work because we have a max limit of 200 https://github.com/AdExNetwork/adex-validator-stack-rust/pull/209/files#diff-3870d7b9b77c57e1688961c8a3f97b50R15

Reply:
Can you move this to the Query and add a check in `is_valid` instead as this hardcoded value on the DB query side can lead to confusion as it's hidden and doesn't give any feedback that it actually limited the value to `200`.